server/src/main/java/org/elasticsearch/common/lucene/search/function/ScriptScoreFunction.java

```diff
@@ -85,9 +110,13 @@ public Explanation explainScore(int docId, Explanation subQueryScore) throws IOE
                 } else {
                     double score = score(docId, subQueryScore.getValue().floatValue());
```

Comment:
What we should do here, is since we know `explainScore` means we are doing an `explain` call, we should handle the logic for creating a `ScoreScript.ExplanationHolder` here.

We should add a new `public double score(int docId, float subQueryScore, ExplanationHolder holder) throws IOException {` that both `score` and `explainScore` use where `explainScore` passes in a non-null holder, and scorer passes in null. This way we only create this object if we are explaining.